### ISSUE TITLE
Empty class when all optional classes are omitted

### ISSUE THREAD

**Issue by SenojLuap** [lambda-fairy/maud#424]
This may be a 'working as intended' situation, but I thought I'd raise the issue in case it's not:

Currently, if you're applying a class to an element optionally, but the condition is falsey, maud still generates a 'class' attribute, but with an empty class list.

Ex.
```rust
use maud::{html, Render};
fn main() {
    let demo_output = html! {
      p .someClass[false] {
        "Lorem Ipsum"
      }
    }
    .render();
    println!("Generated html: {}", demo_output.0);
}
```
Output: `Generated html: <p class="">Lorem Ipsum</p>`

Note that a 'class' attribute is generated, even though it isn't needed.

Definitely not a huge deal, but it would be nice to keep the HTML trim where possible.

**Comment by lambda-fairy** [comment]
Yeah, the current behavior isn't ideal but it's hard to do a clean fix. Consider when there's more than one class – now you need to check that _all_ classes are omitted before removing the attribute. Not to mention, some of the classes might be non-optional too.

I remember there's a related issue about extra spaces, and a complete solution should fix that as well.
